### ISSUE TITLE
Quick Open: Wrong title if underscore in name

### ISSUE THREAD

**Issue by MarcelGerber** [adobe/brackets#10871]
1. In `brackets/src/thirdparty`, copy `CodeMirror2` to `CodeMirror2_`
2. Open Brackets, open Quick Open and type `codemirr`

Result:
The first few entries are named `CodeMirrAUTHORS`, `CodeMirrLICENSE` and so on
![image](https://cloud.githubusercontent.com/assets/2641501/7088367/52fe523a-df93-11e4-8430-103688960875.png)

Expected:
The first few entries are named `AUTHORS`, `LICENSE` and so on

cc @peterflynn

**Comment by peterflynn** [comment]
Repros at least back to 1.1, so it's not a recent regression.  Looks like a bug in StringMatch -- the result items come back like this:

```
{
  label: "AUTHORS",
  stringRanges: [
    {text: "src/thirdparty/", matched: false, includesLastSegment: false},
    {text: "CodeMi", matched: true, includesLastSegment: true},
    {text: "rror2_/AUTHORS", matched: false, includesLastSegment: true}
  ],
  matchGoodness: -1006,
  fullPath: "/Users/pflynn/Brackets/brackets/src/thirdparty/CodeMirror2_/AUTHORS",
  filenameWithoutExtension: "AUTHORS"
}
```

Notice how the middle segment has `includesLastSegment: true` even though it doesn't overlap the last segment (filename) at all.  This messed up the logic in `QuickOpen._filenameResultsFormatter()`'s `fileNameFilter()` helper.

**Comment by MarcelGerber** [comment]
Looks like it's an all-time StringMatch bug. I was able to repro back in Sprint 36.

**Comment by rroshan1** [comment]
Fixed with PR #10878 merge.
